significant_strikes.py

```diff
@@ -0,0 +1,500 @@
+        # Iterate through all combinations to create column names
+        for fighter in fighters:
+            for significant_strike_stat in significant_strike_stats:
+                for round in rounds:
+                    for time_period in time_periods:
+                        col_name = f"{fighter.replace(' ', '_')}_{significant_strike_stat.replace(' ', '_')}_{round}_{time_period.replace(' ', '_')}"
+                        col_names.append(col_name)
+        return col_names
+    
+    def get_fighter_fights_significant_strikes(self, df, fighter_id, index):
+        """
+        Retrieves the most recent fights of a fighter before a given index in the dataframe.
+
+        Args:
+            df (pd.DataFrame): The dataframe containing fight records.
+            fighter_id (str): The ID of the fighter.
+            index (int): The index in the dataframe up to which fights should be considered.
+
+        Returns:
+            tuple: Contains three dataframes - the last 3 fights, the last 5 fights, and all previous fights.
+        """
+
+        all_prev_fights = df.loc[:index-1]
+
+        if not all_prev_fights.empty:
+            # Filter fights involving the specified fighter
+            fighter_a_id_vals = all_prev_fights.fighter_a_id.values
+            fighter_b_id_vals = all_prev_fights.fighter_b_id.values
+
+            prev_fights = all_prev_fights[((fighter_a_id_vals == fighter_id) | (fighter_b_id_vals == fighter_id))]
+            num_fights = len(prev_fights)
+
+            # Initialize empty DataFrames for the scenarios where there are fewer than 3 or 5 fights
+            last_3_fights = pd.DataFrame()
+            last_5_fights = pd.DataFrame()
+
+            # Check if the fighter has at least 3 fights
+            if num_fights >= 3:
+                last_3_fights = prev_fights.tail(3)
+                
+            else:
+                # If fewer than 3 fights, return all available fights
+                last_3_fights = prev_fights
+
+            # Check if the fighter has at least 5 fights
+            if num_fights >= 5:
+                last_5_fights = prev_fights.tail(5)
+                
+            else:
+                # If fewer than 5 fights, return all available fights
+                last_5_fights = prev_fights
+                
+            return last_3_fights, last_5_fights, prev_fights
+
+        return pd.DataFrame(), pd.DataFrame(), pd.DataFrame()
+        
+
+    def get_raw_significant_strikes_stats(self, df, round):
+        """
+        Extracts raw significant strike stats from the dataframe for a specific round.
+
+        Args:
+            df (pd.DataFrame): The dataframe containing fight records.
+            round (str): The round for which the statistics are required.
+
+        Returns:
+            pd.DataFrame: A dataframe with raw stats for the specified round for fighters a and b.
+        """
+
+        # Define column names based on the round
+        if round == "overall":
```

Comment:
String comparisons are costly and as a general coding practice tip, it's better to keep a variable as one type rather than multiple for error handling/debugging purposes. I think in this case it's fine, (you could just make round '0' for the overall case) since it does make the code a bit more readable.